cmd/monaco/main.go

```diff
@@ -44,380 +36,211 @@ func Run(args []string) int {
 }
 
-func containsVersionFlag(args []string) bool {
-	for _, arg := range args {
-		if arg == "--version" || arg == "-version" {
-			return true
-		}
-	}
-
-	return false
-}
-
 func RunImpl(args []string, fileReader util.FileReader) (statusCode int) {
-	if containsVersionFlag(args) {
-		fmt.Println(version.MonitoringAsCode)
-		return 0
-	}
-
-	statusCode = 0
-
-	dryRun, verbose, environments, projectNameToDeploy, path, errorList, flagError := parseInputCommand(args, fileReader)
-
-	if flagError != nil {
-		util.FailOnError(flagError, "could not parse flags")
-	}
+	var app *cli.App
 
-	var deploymentErrors = make(map[string]error)
-
-	for i, err := range errorList {
-		configIssue := fmt.Sprintf("environmentfile-issue-%d", i)
-		deploymentErrors[configIssue] = err
-	}
-
-	err := util.SetupLogging(verbose)
-
-	if err != nil {
-		util.Log.Error("Error writing log file: %s", err.Error())
+	if newCli, ok := os.LookupEnv("NEW_CLI"); ok && newCli != "0" {
+		app = buildExperimentalCli(fileReader)
+	} else {
+		app = buildCli(fileReader)
 	}
 
-	util.Log.Info("Dynatrace Monitoring as Code v" + version.MonitoringAsCode)
-
-	apis := createApis()
+	err := app.Run(args)
 
-	projects, err := project.LoadProjectsToDeploy(projectNameToDeploy, apis, path, fileReader)
 	if err != nil {
-		util.FailOnError(err, "Loading of projects failed")
-	}
-
-	util.Log.Info("Executing projects in this order: ")
-
-	for i, project := range projects {
-		util.Log.Info("\t%d: %s (%d configs)", i+1, project.GetId(), len(project.GetConfigs()))
-	}
-
-	for _, environment := range environments {
-		err := execute(environment, projects, dryRun, path)
-		if err != nil {
-			deploymentErrors[environment.GetId()] = err
-		}
-	}
-
-	for environment, err := range deploymentErrors {
-		if dryRun {
-			util.Log.Error("Validation of %s failed with error %s\n", environment, err)
-		} else {
-			util.Log.Error("Deployment to %s failed with error %s\n", environment, err)
-		}
-		statusCode = -1
-	}
-
-	if statusCode == 0 {
-		if dryRun {
-			util.Log.Info("Validation finished without errors")
-		} else {
-			util.Log.Info("Deployment finished without errors")
-		}
-		err = deleteConfigs(apis, environments, path, dryRun, fileReader)
-		if err != nil {
-			util.Log.Error("Deletion of configs failed with error %s\n", err)
-			statusCode = -1
-		}
+		log.Fatal(err)
+		return 1
 	}
 
-	return statusCode
+	return 0
 }
 
-func parseInputCommand(args []string, fileReader util.FileReader) (dryRun bool, verbose bool, environments map[string]environment.Environment, project string, path string, errorList []error, flagError error) {
-
-	// define flags
-	var environmentsFile string
-	var specificEnvironment string
-	var versionFlag bool
+func buildCli(fileReader util.FileReader) *cli.App {
+	app := cli.NewApp()
 
-	// parse flags
-	shorthand := " (shorthand)"
-	dryRunUsage := "Set dry-run flag to just validate configurations instead of deploying."
+	app.Version = version.MonitoringAsCode
```

Comment:
Don't you need to set `app.Name` here as well?

Reply:
If we don't set the name that seems to result in this: 

```
NAME:
   monaco - A new cli application
```

Is there a way we can get rid of the "A new cli application" without hardcoding the name?

Reply:
otherwise looks no

Reply:
Yes we simply have to set the `Usage`. What shall I set it to?

Reply:
Maybe the following:

> Automates the deployment of Dynatrace Monitoring Configuration to one or multiple Dynatrace environments.

like we have in the repo description?

Reply:
Usage text should be updated now :+1: